test.js

```diff
@@ -180,7 +180,6 @@ function testWriteToStreamError (expected, fixture) {
     const result = mqtt.writeToStream(fixture, stream)
 
     t.false(result, 'result should be false')
-    t.end()
```

Comment:
Basically, this test was failing because `destroy()` emits `error` asynchronously. 

Removing `t.end()` here is fine because the test will automatically end when two assertions are hit and will fail the test if more than 2 are hit (per the [tape documentation](https://github.com/substack/tape#tplann))